### PR TITLE
solved #38 "1.11.3 .stop() does not cancel all functionality" and #37 "1.11.3 .stop() fires callback"

### PR DESCRIPTION
**Pull request by Martin-Eckleben**
This is my first pull request so please be kind :)

I didnt change dist because im not sure if there is a build procedure im not aware of?

I tried to generate docs with npm start as stated in CONTRIBUTING.md but I guess there wasnt much to be documented.

The resetWithCallback solution sure is a budget solution and could probably be better from an architectural view but my time is pretty short and I cant think of a real/bad downside?

If I did anything wrong or could have done it better and you find the time for it I would love to hear about it for learning :)

With best wishes

### CODE REVIEW

**Comment by ThibaultJanBeyer** [comment]
Hey Exxenmann,

Thanks for the pull request, I'm always very happy to see people taking the time to contribute!

I'll have a look next the upcoming week when I'm back from holidays. There may be a simpler solution to the issue.

Until then, thanks a lot!

**Comment by ThibaultJanBeyer** [comment]
@Martin-Eckleben Hey there, did you have a chance to look at it yet?

**Comment by Martin-Eckleben** [comment]
@ThibaultJanBeyer I only had the time to skim your answer yet.
So far already many thanks from my side for your detailed explanations and comments and your time.
Right now I "repaired" my GPG Keys for verified commits exactly before I have to leave now for the bar as a friend from Cologne is visiting :))
I will definitely implement the changes tomorrow morning.
Wish you a great weekend :)

**Comment by ThibaultJanBeyer** [comment]
Okay, looking forward for this. Also wish you a great weekend

File: src/DragSelect.js [hunk 1]
```diff
@@ -727,19 +728,23 @@ class DragSelect {
   // Ending
   //////////////////////////////////////////////////////////////////////////////////////
 
+  resetWithCallback = event => {
+    this.callback(this._selected, event);
+    this.reset();
```
[the remark this answers is not in the document]
Comment (Martin-Eckleben) [reply src/DragSelect.js]: Ah I missed that. Thank you for noticing it and pointing it out!

File: src/DragSelect.js [hunk 2]
```diff
@@ -727,19 +728,23 @@ class DragSelect {
   // Ending
   //////////////////////////////////////////////////////////////////////////////////////
 
+  resetWithCallback = event => {
```
[the remark this answers is not in the document]
Comment (Martin-Eckleben) [reply src/DragSelect.js]: You did all the work here already. Thank you!

File: src/DragSelect.js [hunk 3]
```diff
@@ -782,8 +787,10 @@ class DragSelect {
     this.area.removeEventListener('touchstart', this._startUp, {
       passive: false
     });
-    document.removeEventListener('mouseup', this.reset);
-    document.removeEventListener('touchend', this.reset);
+    document.removeEventListener('mouseup', this.resetWithCallback);
+    document.removeEventListener('touchend', this.resetWithCallback);
+
+    this._handleSelectables(this.selectables,true,true)
```
[the remark this answers is not in the document]
Comment (Martin-Eckleben) [reply src/DragSelect.js]: Done

File: src/DragSelect.js [hunk 4]
```diff
@@ -162,6 +162,7 @@ class DragSelect {
         this._addSelectable(selectable, fromSelection);
       } else if (indexOf > -1 && remove) {
         this._removeSelectable(selectable, indexOf, fromSelection);
+        index--;
```
[the remark this answers is not in the document]
Comment (Martin-Eckleben) [reply src/DragSelect.js]: Ah OK!
Thank you for finding it and pointing it out and already giving the solution.
Hopefully I can be as constructive in the future.

Maybe I would have noticed it by running the automated tests? I found them but am not sure how to run them? I guess calling them manually in the Browser would miss the "automated" part? :)

**Comment by Martin-Eckleben** [comment]
I finally found the time to do it.
Really you did all the work there was!
Thank you very much for your three advice. I will make sure to double check in the future.
Also thank you for "taking me by the hand" in my (a little bit tipsily) first open source steps :)

File: src/DragSelect.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ThibaultJanBeyer) [reply src/DragSelect.js]: `npm i` to install the testing suites and then `npm test` from the terminal should be enough to run the tests.

**Comment by ThibaultJanBeyer** [comment]
Thanks a lot @Martin-Eckleben ! People like you are gold for open source projects 🎉

**Pull request merged**